### PR TITLE
Add percentage on allowance charges

### LINKED ISSUE #429: Add percentage to allowance/charge on outgoing invoices
All allowances and charges should have percentage filed. 
Customer can choose if he would like to populate percentage or amount.
Use JS to recalculate from amount to percentage and opposite .

Issue reported by Arnt Telhaug

### PR DESCRIPTION
**Pull request by Baxxx**
Closes #429

### CODE REVIEW

File: db/changes/167_add_percent_to_allowancecharges.sql [hunk 1]
```diff
@@ -0,0 +1,35 @@
+ALTER TABLE allowancecharge
+ADD COLUMN PercentAmount DECIMAL(16,2) DEFAULT 0,
+ADD COLUMN IsPercentage BOOLEAN DEFAULT FALSE;
```
Comment (Baxxx) [inline db/changes/167_add_percent_to_allowancecharges.sql]: Try without this

File: modules/invoice/view/edit.php [hunk 2]
```diff
@@ -366,6 +366,7 @@ function updateInvoiceAllowanceChargeLineData(element, update_amount) {
   update_amount = typeof update_amount !== 'undefined' ? update_amount : false;
   var id = element.id;
   var name = element.id.split('.');
+  var original_name = element.id.split('.');
```
Comment (Baxxx) [inline modules/invoice/view/edit.php]: ` = name`

File: modules/invoice/view/edit.php [hunk 3]
```diff
@@ -483,7 +542,7 @@ function updateInvoiceLineAllowanceChargeData(element) {
 
   var invoice_line_name = ['invoiceoutline', 'ProductID', invoice_line_id];
   var invoice_line_product_id_element = document.getElementById(invoice_line_name.join('.'));
-  updateInvoiceLineData(invoice_line_product_id_element, false);
+  if(update_line) updateInvoiceLineData(invoice_line_product_id_element, false, false);
```
Comment (Baxxx) [inline modules/invoice/view/edit.php]: Remove third argument

File: modules/invoice/view/edit.php [hunk 4]
```diff
@@ -1097,7 +1189,28 @@ function markRed(elements) {
             'tabindex'  => $tabindex++));
         ?>
       </td>
-      <td colspan="2"></td>
+      <td colspan="2">Percent:
```
Comment (Baxxx) [inline modules/invoice/view/edit.php]: Remove text for now

**Review by sls-lnluksa — APPROVED** [review]

**Comment by sls-lnluksa** [comment]
⚠️  Test his logic! If you have any doubts, schedule testing with Hildegunn.

File: modules/allowancecharge/view/edit.php [hunk 5]
```diff
@@ -75,8 +99,12 @@ function setAllowanceToNegativeAmountIfFirstTime() {
         <td colspan="2"><? print $_lib['form3']->input(array('type'=>'text', 'table'=>$db_table, 'field'=>'Reason', 'pk'=>$row->AllowanceChargeID, 'tabindex'=>$tabindex++, 'value'=>$row->Reason)) ?></td>
     </tr>
     <tr>
+        <td>Percent</td>
```
Comment (Baxxx) [inline modules/allowancecharge/view/edit.php]: `Prosent`

**Comment by Baxxx** [comment]
- [x] Align the percentage fields to the right
- [x] Make the fields a bit wider
- [x] Fix an error when you create a new ac line and try to update the invoice line. Allowancecharge line is not updated.
- [x] Never print Percent

**Comment by stoivo** [comment]
I forgot to add my review, it is good now.

**Pull request merged**